scipy/optimize/_nonlin.py

```diff
@@ -480,7 +480,7 @@ def asjacobian(J):
                         solve=lambda v: solve(J, v),
                         rsolve=lambda v: solve(J.conj().T, v),
                         dtype=J.dtype, shape=J.shape)
-    elif scipy.sparse.isspmatrix(J):
+    elif scipy.sparse.issparse(J):
```

Comment:
This should probably be reverted, at least for now: the code below is relying on `J*v` being a matrix-vector product, so a sparse array isn't a viable substitute here.